src/elements/CodemodList.ts

```diff
@@ -0,0 +1,239 @@
+	materialUI5: 'intuita.executeMuiCodemods',
+	redwoodJS4: 'intuita.executeRedwoodJsCore4Codemods',
+	reactRouterDom6: 'intuita.executeReactRouterv6Codemods',
+	immutableJSV4: 'intuita.executeImmutableJSv4Codemods',
+	immutableJS0: 'intuita.executeImmutableJSv0Codemods',
+};
+
+// TODO: get this from an API
+const packageUpgradeList: PackageUpgradeItem[] = [
+	{
+		id: 'next13',
+		packageName: 'next',
+		name: 'Next13 Codemods',
+		kind: 'upgrade',
+		leastVersionSupported: '13.0.0',
+		latestVersionSupported: '13.2.4',
+		leastSupportedUpgrade: '12.0.0',
+	},
+	{
+		id: 'next13experimental',
+		packageName: 'next',
+		name: 'Next13 Experimental Codemods',
+		kind: 'upgrade',
+		leastVersionSupported: '13.0.0',
+		latestVersionSupported: '13.2.4',
+		leastSupportedUpgrade: '12.0.0',
+	},
+	{
+		id: 'materialUI5',
+		packageName: '@material-ui/core',
+		name: 'MaterialUI5 Codemods',
+		kind: 'migration',
+		leastVersionSupported: '5.0.0',
+		latestVersionSupported: '5.11.15',
+		leastSupportedUpgrade: '4.0.0',
+	},
+	{
+		id: 'redwoodJS4',
+		packageName: '@redwoodjs/core',
+		name: 'RedwoodJS4 Codemods',
+		kind: 'migration',
+		leastVersionSupported: '4.0.0',
+		latestVersionSupported: '4.0.0',
+		leastSupportedUpgrade: '3.0.0',
+	},
+	{
+		id: 'reactRouterDom6',
+		packageName: 'react-router-dom',
+		name: 'ReactRouterDom6 Codemods',
+		kind: 'migration',
+		leastVersionSupported: '6.0.0',
+		latestVersionSupported: '6.9.0',
+		leastSupportedUpgrade: '5.0.0',
+	},
+	{
+		id: 'immutableJSV4',
+		packageName: 'immutable',
+		name: 'ImmutableJS4 Codemods',
+		kind: 'upgrade',
+		leastVersionSupported: '4.0.0',
+		latestVersionSupported: '4.0.0',
+		leastSupportedUpgrade: '3.0.0',
+	},
+	{
+		id: 'immutableJS0',
+		packageName: 'immutable',
+		name: 'ImmutableJS4 removal Codemods',
+		kind: 'remove',
+		leastVersionSupported: '4.0.0',
+		latestVersionSupported: '4.0.0',
+		leastSupportedUpgrade: '3.0.0',
+	},
+];
+
+const buildCodemodItemHash = (codemodItem: CodemodItem) => {
+	return buildHash(
+		`${codemodItem.label} ${codemodItem.id} ${codemodItem.commandToExecute}`,
+	) as ElementHash;
+};
+
+class CodemodItem extends TreeItem {
+	readonly kind: string;
+	constructor(
+		public readonly label: string,
+		public readonly description: string,
+		public readonly collapsibleState: TreeItemCollapsibleState,
+		public readonly commandToExecute: string,
+	) {
+		super(label, collapsibleState);
+		this.description = description;
+		this.tooltip = `${label}-${description}`;
+		this.kind = 'codemodItem';
+		this.contextValue = 'codemodItem';
+		this.commandToExecute = commandToExecute;
+	}
+	iconPath = path.join(
+		__filename,
+		'..',
+		'..',
+		'resources',
+		'bluelightbulb.svg',
+	);
+}
+
+class CodemodTreeProvider {
+	rootPath: string | null | undefined;
+	#messageBus: MessageBus;
+	#codemodItemsMap: Map<ElementHash, CodemodItem> = new Map();
+
+	constructor(path: string | null | undefined, messageBus: MessageBus) {
```

Comment:
`string | null` IMHO, based on the observation that:
* `null` means that the `path` is set not to contain any value on purpose
* `undefined` means the `path` is undefined but you don't know why that is